DataStructure/LinkList/linklist.java

```diff
@@ -0,0 +1,455 @@
+package LinkList;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class linklist {
+	// Create Node
+	class Node {
+		int data;
+		Node next;
+
+		Node(int val) {
+			this.data = val;
+			this.next = null;
+		}
+
+	}
+
+	private Node head = null;
+	private Node tail = null;
+
+	// Insertion in List
+	void insertiontolast(int data) {
+		if (head == null) {
+			Node temp = new Node(data);
```

Comment:
line 21 and 31 repeated code. The line can be outside the if else block. Can be written only once.